libecs/source/Core/Component/ScriptingComponentArray.cxx

```diff
@@ -0,0 +1,20 @@
+#include "pivot/ecs/Core/Component/ScriptingComponentArray.hxx"
+
+namespace pivot::ecs::component
+{
+
+const Description &ScriptingComponentArray::getDescription() const { return m_description; }
+
+std::optional<data::Value> ScriptingComponentArray::getValueForEntity(Entity entity) const
+{
+    auto value = m_components.find(entity);
+    if (value == m_components.end()) return std::nullopt;
+    return value->second;
+}
+
+void ScriptingComponentArray::setValueForEntity(Entity entity, std::optional<data::Value> value)
+{
+    m_components[entity] = value.value();
```

Comment:
If `entity` is not in `m_components` is this an error ? Because right now, it will add to the array.

If it is supposed to be an error :
```suggestion
    m_components.at(entity) = value.value();
```